test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+          status: "active",
+        };
+        const resourceKey = uuid.v4();
+        const resourceHref = type + "/" + resourceKey;
+        const attachmentKey1 = uuid.v4();
+        const attachmentKey2 = uuid.v4();
+
+        const responsePut = await httpClient.put({ path: resourceHref, body });
+        assert.equal(responsePut.status, 201);
+
+        // Add attachment
+        debug("PUTting the profile image as attachment");
+        const filesToPut = [
+          {
+            remotefileName: "profile.png",
+            localFilename: "test/images/orange-boy-icon.png",
+            attachmentKey: attachmentKey1,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut);
+
+        debug("Adding another profile picture");
+        const filesToPut2 = [
+          {
+            remotefileName: "profile2.png",
+            localFilename: "test/images/avatar-black.png",
+            attachmentKey: attachmentKey2,
+            resourceHref,
+          },
+        ];
+
+        await uploadFilesAndCheck(httpClient, filesToPut2);
+
+        // Check if we have the two expected attachments
+        const responseGet1 = await httpClient.get({ path: type + "/" + resourceKey });
+        assert.equal(responseGet1.status, 200);
+        console.log(responseGet1.body.attachments.length, 2);
+        for (const href of responseGet1.body.attachments.map((a) => a.href)) {
+          const responseGetA = await httpClient.get({ path: href });
+          assert.equal(responseGetA.status, 200);
+        }
+
+        debug("Replacing one of two attachments");
+        const filesToPut3 = [
+          {
+            remotefileName: "profile1.png",
+            localFilename: "test/images/avatar-blue.png",
+            attachmentKey: attachmentKey1,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut3);
+
+        // Check if (only) the two expected attachments are there
+        const responseGet2 = await httpClient.get({ path: type + "/" + resourceKey });
+        assert.equal(responseGet2.status, 200);
+        console.log(responseGet2.body.attachments.length, 2);
+
+        const getStream1 = await doGetStream(
+          httpClient,
+          `/partiesS3/${resourceKey}/attachments/profile1.png`
+        );
+
+        checkStreamEqual(
+          getStream1,
+          fs.createReadStream("test/images/avatar-blue.png")
+        );
+
+        const getStream2 = await doGetStream(
+          httpClient, 
+          `/partiesS3/${resourceKey}/attachments/profile2.png`
+        );
+
+        checkStreamEqual(
+          getStream2,
+          fs.createReadStream("test/images/avatar-black.png")
+        );
+
+        const responseGet3 = await httpClient.get({
+          path: `/partiesS3/${resourceKey}/attachments/profile.png`
+        });
+        // TODO: this does not work: attachment is not being overwritten by reusing a attachmentKey !
```

Comment:
Overwriting in S3 is done by using the same filename (which is already checked in the first test). So I guess this test can be deleted.

Reply:
New cases for attachments replacement are added.